File: API/_archive/GraphQL/python/usecase3_mutations.py
import strawberry
import asyncio

@strawberry.type
class User:
    id: str
    name: str

@strawberry.type
class Query:
    @strawberry.field
    def dummy(self) -> str:
        return "GraphQL requires at least one query"

@strawberry.type
class Mutation:
    @strawberry.mutation
    def create_user(self, name: str) -> User:
        # Simulated DB Insert
        print(f"Inserting {name} into database...")
        return User(id="generated_uuid_123", name=name)

schema = strawberry.Schema(query=Query, mutation=Mutation)

async def run_mutation():
    mutation = """
        mutation {
            createUser(name: "Bob") {
                id
                name
            }
        }
    """
    result = await schema.execute(mutation)
    print("Mutation Result:", result.data)

if __name__ == "__main__":
    asyncio.run(run_mutation())

File: API/_archive/GraphQL/golang/usecase3_mutations.go
package main

import (
	"fmt"
	"github.com/graphql-go/graphql"
)

var userType = graphql.NewObject(graphql.ObjectConfig{
	Name: "User",
	Fields: graphql.Fields{
		"id":   &graphql.Field{Type: graphql.String},
		"name": &graphql.Field{Type: graphql.String},
	},
})

var rootMutation = graphql.NewObject(graphql.ObjectConfig{
	Name: "RootMutation",
	Fields: graphql.Fields{
		"createUser": &graphql.Field{
			Type: userType,
			Args: graphql.FieldConfigArgument{
				"name": &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
			},
			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
				name := p.Args["name"].(string)
				// Simulated DB Insert
				newUser := map[string]string{
					"id":   "generated_uuid_123",
					"name": name,
				}
				return newUser, nil
			},
		},
	},
})

var rootQuery = graphql.NewObject(graphql.ObjectConfig{
	Name: "RootQuery",
	Fields: graphql.Fields{
		"dummy": &graphql.Field{Type: graphql.String},
	},
})

func main() {
	schema, _ := graphql.NewSchema(graphql.SchemaConfig{
		Query:    rootQuery,
		Mutation: rootMutation,
	})

	query := `mutation { createUser(name: "Bob") { id name } }`
	
	result := graphql.Do(graphql.Params{
		Schema:        schema,
		RequestString: query,
	})

	fmt.Printf("Mutation Result: %v\n", result.Data)
}
